src/index.ts

```diff
@@ -59,23 +59,22 @@ export type NoDefaultRec<Val> = {
  * @param valProp An optional custom name for the value property of the union. If not specified,
  * the value must be a dictionary type.
  */
-export function unionize<Record extends MultiValueRec>(
-  record: Record,
-): Unionized<Record, MultiValueVariants<Record, 'tag'>>;
-export function unionize<Record extends MultiValueRec, TagProp extends string>(
-  record: Record,
-  tagProp: TagProp,
-): Unionized<Record, MultiValueVariants<Record, TagProp>>;
+
 export function unionize<
   Record extends SingleValueRec,
   TagProp extends string,
   ValProp extends string
 >(
   record: Record,
-  tagProp: TagProp,
-  valProp: ValProp,
+  config: { valProp: ValProp; tagProp?: TagProp },
 ): Unionized<Record, SingleValueVariants<Record, TagProp, ValProp>>;
-export function unionize<Record>(record: Record, tagProp = 'tag', valProp?: string) {
+export function unionize<Record extends MultiValueRec, TagProp extends string>(
+  record: Record,
+  config?: { tagProp: TagProp },
+): Unionized<Record, MultiValueVariants<Record, TagProp>>;
+export function unionize<Record>(record: Record, config?: { valProp?: string; tagProp?: string }) {
```

Comment:
Now that these names are being exposed as part of the API, I'd probably vote to call them `tag` and `value` respectively.

Reply:
Ok makes sense. I like tag & value more. Although 'data' might be a good name too.

Reply:
I'm fine with `value`, `data` or `payload`.